.github/workflows/python-publish.yml

```diff
@@ -13,18 +13,25 @@ jobs:
 
     steps:
       - uses: actions/checkout@v3
-      - name: Set up Python
-        uses: actions/setup-python@v3
+      - uses: actions/setup-python@v3
         with:
           python-version: "3.x"
-      - name: Install dependencies
+
+      - name: install build package
         run: |
-          python -m pip install --upgrade pip
-          pip install setuptools wheel twine
-      - name: Build and publish
+          pip install --upgrade pip
+          pip install build
+          pip freeze
+
+      - name: build release
+        run: |
+          python -m build --sdist --wheel .
+          ls -l dist
+
+      - name: Publish to PyPI
         env:
-          TWINE_USERNAME: "__token__"
+          TWINE_USERNAME: __token__
           TWINE_PASSWORD: ${{ secrets.PYPI_TOKEN }}
         run: |
-          python setup.py sdist bdist_wheel
-          twine upload dist/*
+          pip install twine
+          twine upload --skip-existing dist/*
```

Comment:
It is my understanding that use of `setup.py sdist bdist_wheel` is deprecated, and we are supposed to start using a tool like `build`. This is what I've picked up from MinRK which seem to match what python.org documents:

- https://packaging.python.org/en/latest/tutorials/packaging-projects/#generating-distribution-archives
- https://peps.python.org/pep-0517/